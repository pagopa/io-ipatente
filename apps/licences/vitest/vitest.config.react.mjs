/// <reference types="vitest" />
/// <reference types="vite/client" />

import react from "@vitejs/plugin-react";
import { loadEnv } from "vite";
import tsconfigPaths from "vite-tsconfig-paths";
import { configDefaults, defineConfig } from "vitest/config";

export default defineConfig({
  plugins: [react(), tsconfigPaths()],
  test: {
    coverage: {
      exclude: [
        ...configDefaults.coverage.exclude,
        "**/__tests__",
        "src/generated/**",

        // BE
        "src/app/*",
        "src/config/*",
        "src/lib/bff/*",
      ],
      ignoreEmptyLines: true,
      include: ["src/**/*.{tsx,ts}"],
      reporter: ["text", "html"],
    },
    css: true,
    env: loadEnv("test", process.cwd(), ""),
    environment: "jsdom",
    exclude: [
      ...configDefaults.exclude,
      "./vitest/vitest.setup.react.ts",
      "./vitest/vitest.setup.node.ts",
    ],
    globals: true,
    include: ["./src/**/*.test.tsx"],
    setupFiles: "./vitest/vitest.setup.react.ts",
    typecheck: {
      ignoreSourceErrors: true,
    },
  },
});
