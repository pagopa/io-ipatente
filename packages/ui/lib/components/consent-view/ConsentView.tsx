import { Check } from "@mui/icons-material";
import { Box, Button, Stack, Typography } from "@mui/material";

export interface ConsentViewProps {
  /** Labels for bottm action buttons */
  actionLabels: {
    /** Label for consent action */
    consent: string;
    /** Label for decline consent action */
    dissent: string;
  };
  /** Consent View main description */
  description: string;
  /** More information text */
  moreInfo: string;
  /** Event triggered when user click on consent action */
  onConsentClick: () => void;
  /** Event triggered when user click on dissent action */
  onDissentClick: () => void;
  /** Consent data section */
  requiredData: {
    /** List of user data for which consent is required */
    data: string[];
    /** Consent section title */
    title: string;
  };
  /** Consent View main title */
  title: string;
}

/** Operation result screen component */
export const ConsentView = ({
  actionLabels,
  description,
  moreInfo,
  onConsentClick,
  onDissentClick,
  requiredData,
  title,
}: ConsentViewProps) => (
  <Box display="flex" flex={1} flexDirection="column">
    {/* Scrollable content */}
    <Box flex={1} overflow="auto">
      <Stack paddingY={1} spacing={1}>
        <Typography fontWeight={700} variant="h5">
          {title}
        </Typography>
        <Typography variant="body2">
          <span
            dangerouslySetInnerHTML={{
              __html: description,
            }}
          />
        </Typography>
        <Stack paddingBottom={2} paddingTop={3} spacing={1}>
          <Typography variant="overline">{requiredData.title}</Typography>
          {requiredData.data.map((value, index) => (
            <Stack
              alignItems="center"
              bgcolor="#FAFAFA"
              direction="row"
              justifyContent="space-between"
              key={`required-data-${index}`}
              padding={1.5}
            >
              <Typography fontWeight={600} variant="body2">
                {value}
              </Typography>
              <Check></Check>
            </Stack>
          ))}
        </Stack>
        <Typography color="text.secondary" fontSize="14px" fontWeight={400}>
          <span
            dangerouslySetInnerHTML={{
              __html: moreInfo,
            }}
          />
        </Typography>
      </Stack>
    </Box>
    {/* Sticky bottom actions */}
    <Box
      bgcolor="background.default"
      bottom={0}
      paddingTop={2}
      position="sticky"
      textAlign="center"
    >
      <Stack alignItems="stretch" bottom={0} direction="column" spacing={1}>
        <Button onClick={onConsentClick} variant="contained">
          {actionLabels.consent}
        </Button>
        <Button onClick={onDissentClick} variant="text">
          {actionLabels.dissent}
        </Button>
      </Stack>
    </Box>
  </Box>
);
