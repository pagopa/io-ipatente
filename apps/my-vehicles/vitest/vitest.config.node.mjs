import { configDefaults, defineConfig } from "vitest/config";
import tsconfigPaths from "vite-tsconfig-paths";

export default defineConfig({
  plugins: [tsconfigPaths()],
  test: {
    coverage: {
      exclude: [
        ...(configDefaults.coverage.exclude
          ? configDefaults.coverage.exclude
          : []),
        ".next/*",
        "mocks/*",
        "public/*",
        "*.config.js",
        "src/generated/**",
        "src/config/*",
        "src/types/*",
        "src/main.ts",
        "src/config.ts",
        "src/instrumentation.ts",
        "**/__mocks__/**",
        // FE
        "src/hooks/*",
        "src/layouts/*",
      ],
      extension: configDefaults.coverage.extension
        ?.toString()
        .replace(",.tsx", "")
        .split(","),
      ignoreEmptyLines: true,
      reporter: ["text", "json-summary", "json"],
      reportsDirectory: `${configDefaults.coverage.reportsDirectory}/backend`,
    },
    exclude: [
      ...configDefaults.exclude,
      "./vitest/vitest.setup.react.ts",
      "./vitest/vitest.setup.node.ts",
    ],
    include: ["./src/**/__tests__/*.ts"],
    setupFiles: ["./vitest/vitest.setup.node.ts"],
    typecheck: {
      ignoreSourceErrors: true,
    },
  },
});
