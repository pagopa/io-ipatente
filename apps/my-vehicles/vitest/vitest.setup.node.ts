/**
 * MSW configuration for Vitest (used as setupFile in vitest.config.node.ts)
 */
import { loadEnvConfig } from "@next/env";
import { afterAll, afterEach, beforeAll } from "vitest";

import { mswServerTest } from "../mocks/msw-server";

let mswServer: ReturnType<typeof mswServerTest>;
// Start server before all tests
beforeAll(() => {
  loadEnvConfig(process.cwd());
  // this done to correctly load the environment variables on msw handlers
  mswServer = mswServerTest();
  mswServer.listen({ onUnhandledRequest: "warn" });
});

//  Close server after all tests
afterAll(() => mswServer.close());

// Reset handlers after each test `important for test isolation`
afterEach(() => mswServer.resetHandlers());
