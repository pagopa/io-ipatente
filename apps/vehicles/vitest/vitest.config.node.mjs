import { loadEnv } from "vite";
import tsconfigPaths from "vite-tsconfig-paths";
import { configDefaults, defineConfig } from "vitest/config";

export default defineConfig({
  plugins: [tsconfigPaths()],
  test: {
    coverage: {
      exclude: [
        ...configDefaults.coverage.exclude,
        "**/__tests__",
        "src/generated/**",
        "src/app/api/auth/mocks/**",
        // FE
        "src/common/*",
        "src/hooks/*",
      ],
      ignoreEmptyLines: true,
      include: ["src/**/*.ts"],
      reporter: ["text", "html"],
      reportsDirectory: `${configDefaults.coverage.reportsDirectory}/backend`,
    },
    env: loadEnv("test", process.cwd(), ""),
    exclude: [
      ...configDefaults.exclude,
      "./vitest/vitest.setup.react.ts",
      "./vitest/vitest.setup.node.ts",
    ],
    include: ["./src/**/__tests__/*.ts"],
    setupFiles: ["./vitest/vitest.setup.node.ts"],
    typecheck: {
      ignoreSourceErrors: true,
    },
  },
});
