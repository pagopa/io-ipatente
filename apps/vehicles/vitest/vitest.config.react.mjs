/// <reference types="vitest" />
/// <reference types="vite/client" />

import react from "@vitejs/plugin-react";
import tsconfigPaths from "vite-tsconfig-paths";
import { configDefaults, defineConfig } from "vitest/config";
import { loadEnv } from "vite";

export default defineConfig({
  plugins: [react(), tsconfigPaths()],
  test: {
    coverage: {
      exclude: [
        ...(configDefaults.coverage.exclude
          ? configDefaults.coverage.exclude
          : []),
        ".next/*",
        "mocks/*",
        "public/*",
        "*.config.js",
        "src/generated/**",
        "src/config/*",
        "src/types/*",
        "src/main.ts",
        "src/config.ts",
        "src/instrumentation.ts",
        "**/__mocks__/**",
        // BE
        "src/app/*",
        "src/lib/be/*",
      ],
      ignoreEmptyLines: true,
      reporter: ["text", "json-summary", "json"],
      reportsDirectory: `${configDefaults.coverage.reportsDirectory}/frontend`,
    },
    css: true,
    env: loadEnv("test", process.cwd(), ""),
    environment: "jsdom",
    exclude: [
      ...configDefaults.exclude,
      "./vitest/vitest.setup.react.ts",
      "./vitest/vitest.setup.node.ts",
    ],
    globals: true,
    include: ["./src/**/*.test.tsx"],
    setupFiles: "./vitest/vitest.setup.react.ts",
    typecheck: {
      ignoreSourceErrors: true,
    },
  },
});
