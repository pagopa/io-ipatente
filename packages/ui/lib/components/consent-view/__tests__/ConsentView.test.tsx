import { fireEvent, render } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";

import { ConsentView, ConsentViewProps } from "../ConsentView";

const mockConsentClick = vi.fn();
const mockDissentClick = vi.fn();

const defaultProps: ConsentViewProps = {
  actionLabels: {
    consent: "consent-label",
    dissent: "dissent-label",
  },
  description: "description",
  moreInfo: "more-info-text",
  onConsentClick: mockConsentClick,
  onDissentClick: mockDissentClick,
  requiredData: {
    data: ["required-data-1", "required-data-2", "required-data-3"],
    title: "required-data-title",
  },
  title: "title",
};

describe("Test ConsentView Component", () => {
  it("Should match the snapshot with all props", () => {
    const comp = render(<ConsentView {...defaultProps} />);
    expect(comp).toMatchSnapshot();
  });

  it("Should call onClick when the action buttons are clicked", async () => {
    const comp = render(<ConsentView {...defaultProps} />);

    const consentButton = comp.getByRole("button", {
      name: /consent-label/i,
    });
    const dissentButton = comp.getByRole("button", {
      name: /dissent-label/i,
    });

    fireEvent.click(consentButton);
    fireEvent.click(dissentButton);

    expect(mockConsentClick).toHaveBeenCalledTimes(1);
    expect(mockDissentClick).toHaveBeenCalledTimes(1);
  });
});
