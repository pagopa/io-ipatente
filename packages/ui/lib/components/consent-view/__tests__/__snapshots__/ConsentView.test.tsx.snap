// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test ConsentView Component > Should match the snapshot with all props 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-1kdy0wu"
      >
        <div
          class="MuiBox-root css-4z6nar"
        >
          <div
            class="MuiStack-root css-h4grc4-MuiStack-root"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-bezmlo-MuiTypography-root"
            >
              title
            </h5>
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              <span>
                description
              </span>
            </p>
            <div
              class="MuiStack-root css-1etjgou-MuiStack-root"
            >
              <span
                class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
              >
                required-data-title
              </span>
              <div
                class="MuiStack-root css-1tscs96-MuiStack-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
                >
                  required-data-1
                </p>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CheckIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                  />
                </svg>
              </div>
              <div
                class="MuiStack-root css-1tscs96-MuiStack-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
                >
                  required-data-2
                </p>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CheckIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                  />
                </svg>
              </div>
              <div
                class="MuiStack-root css-1tscs96-MuiStack-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
                >
                  required-data-3
                </p>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CheckIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                  />
                </svg>
              </div>
            </div>
            <p
              class="MuiTypography-root MuiTypography-body1 css-dkseb6-MuiTypography-root"
            >
              <span>
                more-info-text
              </span>
            </p>
          </div>
        </div>
        <div
          class="MuiBox-root css-1ap3phj"
        >
          <div
            class="MuiStack-root css-54e9ld-MuiStack-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              consent-label
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              dissent-label
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-1kdy0wu"
    >
      <div
        class="MuiBox-root css-4z6nar"
      >
        <div
          class="MuiStack-root css-h4grc4-MuiStack-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-bezmlo-MuiTypography-root"
          >
            title
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            <span>
              description
            </span>
          </p>
          <div
            class="MuiStack-root css-1etjgou-MuiStack-root"
          >
            <span
              class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
            >
              required-data-title
            </span>
            <div
              class="MuiStack-root css-1tscs96-MuiStack-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
              >
                required-data-1
              </p>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                />
              </svg>
            </div>
            <div
              class="MuiStack-root css-1tscs96-MuiStack-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
              >
                required-data-2
              </p>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                />
              </svg>
            </div>
            <div
              class="MuiStack-root css-1tscs96-MuiStack-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
              >
                required-data-3
              </p>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                />
              </svg>
            </div>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 css-dkseb6-MuiTypography-root"
          >
            <span>
              more-info-text
            </span>
          </p>
        </div>
      </div>
      <div
        class="MuiBox-root css-1ap3phj"
      >
        <div
          class="MuiStack-root css-54e9ld-MuiStack-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            consent-label
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            dissent-label
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
